File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/widget/HackyRtlViewPager.java
package org.cru.godtools.tract.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import com.duolingo.open.rtlviewpager.RtlViewPager;

import org.ccci.gto.android.common.util.view.ViewUtils;

public class HackyRtlViewPager extends RtlViewPager {
    public HackyRtlViewPager(final Context context) {
        super(context);
    }

    public HackyRtlViewPager(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onInterceptTouchEvent(final MotionEvent ev) {
        try {
            return super.onInterceptTouchEvent(ev);
        } catch (final RuntimeException e) {
            return ViewUtils.handleOnInterceptTouchEventException(e);
        }
    }

    @Override
    public boolean onTouchEvent(final MotionEvent ev) {
        try {
            return super.onTouchEvent(ev);
        } catch (final RuntimeException e) {
            return ViewUtils.handleOnTouchEventException(e);
        }
    }
}

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/widget/HackyRtlViewPager.kt
package org.cru.godtools.tract.widget

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import com.duolingo.open.rtlviewpager.RtlViewPager
import org.ccci.gto.android.common.util.view.ViewUtils

class HackyRtlViewPager @JvmOverloads constructor(context: Context?, attrs: AttributeSet? = null) :
    RtlViewPager(context, attrs) {
    override fun onInterceptTouchEvent(ev: MotionEvent) = try {
        super.onInterceptTouchEvent(ev)
    } catch (e: RuntimeException) {
        ViewUtils.handleOnInterceptTouchEventException(e)
    }

    override fun onTouchEvent(ev: MotionEvent) = try {
        super.onTouchEvent(ev)
    } catch (e: RuntimeException) {
        ViewUtils.handleOnTouchEventException(e)
    }
}
